### PR TITLE
feat: adding check for go mod tidy

### PR DESCRIPTION
**Pull request by asishrs**

### CODE REVIEW

File: commands/go-mod-tidy.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"github.com/spf13/cobra"
+)
+
+var goModTidy = &cobra.Command{
+	Use:   "go-mod-tidy",
+	Short: "Prune no-longer required commits from go.mod",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		runTool("go", append([]string{"mod", "tidy"}))
```
Comment (lietu) [inline commands/go-mod-tidy.go]: Running `pre-commit run --all-files` on the repo itself complains about this line:

```
commands\go-mod-tidy.go:12:17: SA4021: x = append(y) is equivalent to x = y (staticcheck)
                runTool("go", append([]string{"mod", "tidy"}))
```

Compared to many of the other commands the main difference is that it does not include `args...`. It does not seem `go mod tidy` currently accepts all that many arguments, but there is *an* argument it accepts, and we could support it and future arguments by including `args...`.

Changing the line to:

```
runTool("go", append([]string{"mod", "tidy"}, args...))
```

Should fix the issue.

**Review by lietu — CHANGES_REQUESTED** [review]
Thanks for the PR!

I took a look and ran `pre-commit run --all-files` locally and noticed there were a few things it was complaining about, some small automatic edit to `README.md` and:

```
commands\go-mod-tidy.go:12:17: SA4021: x = append(y) is equivalent to x = y (staticcheck)
                runTool("go", append([]string{"mod", "tidy"}))
```

Could you please fix these? :)

**Comment by lietu** [comment]
Also when I run `go mod tidy` locally with your branch I get a number of changes to `go.mod` and `go.sum` - maybe you need to run it again?

File: commands/go-mod-tidy.go (hunk 1 above)
Comment (asishrs) [reply commands/go-mod-tidy.go]: Sorry for that. fixed it

**Comment by lietu** [comment]
Thanks for the contribution!

**Pull request merged**